lib/components/TagsInput/TagsInput.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from 'react';
+import { TagsInput as ArkTagsInput, TagsInputValueChangeDetails as ArkValueChangeDetails } from '@ark-ui/react';
+
+export interface TagsInputProps {
+  label: string;
+  placeholder: string;
+  tags: string[];
+  onValueChange: (values: string[]) => void;
+}
+
+export const TagsInput = ({ label, placeholder, tags, onValueChange }: TagsInputProps) => {
+  const [selectedTag, setSelectedTag] = useState<string | null>(null);
+
+  const highlightChange = (details: ArkTagsInput.HighlightChangeDetails) => {
+    setSelectedTag(details.highlightedValue);
+  };
+
+  const onValueChangeHandler = (details: ArkValueChangeDetails) => {
+    onValueChange(details.value);
+  };
+
+  return (
+    <ArkTagsInput.Root
+      className="flex flex-col"
+      value={tags}
+      onHighlightChange={highlightChange}
+      onValueChange={onValueChangeHandler}
+      validate={(details) => {
+        return !details.value.includes(details.inputValue);
+      }}
+      onFocusOutside={() => setSelectedTag(null)}
+      onBlur={() => setSelectedTag(null)}
+    >
+      <ArkTagsInput.Context>
+        {(tagsInput) => (
+          <>
+            <ArkTagsInput.Label className="mb-4 text-form-label text-primary-gray">{label}</ArkTagsInput.Label>
+            <div className="rounded-[10px] border-[5px] border-border-gray p-5">
+              <ArkTagsInput.Input placeholder={placeholder && `(${placeholder})`} className="mb-3 w-full" />
+              <ArkTagsInput.Control className="flex gap-x-3">
+                {tagsInput.value.map((value, index) => (
+                  <ArkTagsInput.Item
+                    key={index}
+                    index={index}
+                    value={value}
+                    className={`${selectedTag == value ? 'outline outline-2 outline-offset-[1.5px]' : ''} group inline-flex select-none items-center justify-center rounded-sm border-2 border-secondary-gray px-4 text-tag`.trim()}
+                  >
+                    <ArkTagsInput.ItemPreview className="gap-3">
+                      <div className="inline-flex items-center">
```

Comment:
This `div` is added so that the `hidden` attribute that Ark adds, will work (giving any `display` value will override `hidden` attribute).